django/views/decorators/cache.py

```diff
@@ -1,10 +1,24 @@
 from functools import wraps
 
+from django.http import HttpRequest
 from django.middleware.cache import CacheMiddleware
 from django.utils.cache import add_never_cache_headers, patch_cache_control
 from django.utils.decorators import decorator_from_middleware_with_args
 
 
+def ensure_http_request(maybe_request):
+    """
+    Check that the argument is an HttpRequest, and fail
+    with a helpfull error if it's not.
+    """
+    if not isinstance(maybe_request, HttpRequest):
+        raise TypeError(
+            "never_cache didn't receive an HttpRequest. "
```

Comment:
This is not a correct message for `@cache_control`, I'm going to move this check to decorators.